File: crates/normalize-languages/tests/fixtures/swift/sample.swift
import Foundation
import Swift

struct Point {
    let x: Double
    let y: Double

    func distanceTo(_ other: Point) -> Double {
        let dx = x - other.x
        let dy = y - other.y
        return (dx * dx + dy * dy).squareRoot()
    }
}

class Stack<T> {
    private var items: [T] = []

    func push(_ item: T) {
        items.append(item)
    }

    func pop() -> T? {
        if items.isEmpty {
            return nil
        }
        return items.removeLast()
    }

    func peek() -> T? {
        return items.last
    }

    var isEmpty: Bool {
        return items.isEmpty
    }

    var count: Int {
        return items.count
    }
}

func classify(_ n: Int) -> String {
    if n < 0 {
        return "negative"
    } else if n == 0 {
        return "zero"
    } else {
        return "positive"
    }
}

func sumEvens(_ numbers: [Int]) -> Int {
    var total = 0
    for n in numbers {
        if n % 2 == 0 {
            total += n
        }
    }
    return total
}

let stack = Stack<Int>()
stack.push(10)
stack.push(20)
print(stack.pop() ?? 0)
print(classify(-5))
print(sumEvens([1, 2, 3, 4, 5]))

File: crates/normalize-languages/tests/fixtures/kotlin/sample.kt
import java.util.LinkedList
import kotlin.math.abs

data class Point(val x: Double, val y: Double) {
    fun distanceTo(other: Point): Double {
        val dx = x - other.x
        val dy = y - other.y
        return Math.sqrt(dx * dx + dy * dy)
    }
}

class Queue<T> {
    private val items = LinkedList<T>()

    fun enqueue(item: T) {
        items.addLast(item)
    }

    fun dequeue(): T? {
        return if (items.isEmpty()) null else items.removeFirst()
    }

    fun peek(): T? = items.peekFirst()

    val size: Int get() = items.size
}

// Classify a number
@JvmStatic
fun classify(n: Int): String {
    return when {
        n < 0 -> "negative"
        n == 0 -> "zero"
        else -> "positive"
    }
}

fun sumEvens(numbers: List<Int>): Int {
    var total = 0
    for (n in numbers) {
        if (n % 2 == 0) {
            total += n
        }
    }
    return total
}

fun main() {
    val q = Queue<Int>()
    q.enqueue(1)
    q.enqueue(2)
    println(q.dequeue())
    println(classify(-3))
    println(sumEvens(listOf(1, 2, 3, 4, 5)))
    val p1 = Point(0.0, 0.0)
    val p2 = Point(3.0, 4.0)
    println(p1.distanceTo(p2))
}
